contracts/SDCollateral.sol

```diff
@@ -80,6 +81,32 @@ contract SDCollateral is ISDCollateral, AccessControlUpgradeable, ReentrancyGuar
         slashSD(operator, sdToSlash);
     }
 
+    /// @notice slashes one validator equi. SD amount
+    /// @dev callable only by respective withdrawVaults
+    /// @param _validatorId validator SD collateral to slash
+    function slashSSVOperatorSD(
+        uint8 _poolId,
+        uint256 _validatorId,
+        uint64[] memory operatorIds
+    ) external override nonReentrant {
+        (, , , , address withdrawVaultAddress, , , ) = INodeRegistry(
+            IPoolUtils(staderConfig.getPoolUtils()).getNodeRegistry(_poolId)
+        ).validatorRegistry(_validatorId);
+        if (msg.sender != withdrawVaultAddress) {
+            revert CallerNotWithdrawVault();
+        }
+        isPoolThresholdValid(_poolId);
+        uint256 sdToSlash = convertETHToSD(poolThresholdbyPoolId[_poolId].minThreshold);
+        for (uint8 i; i < operatorIds.length; i++) {
```

Comment:
gas optimization: use `uint256` for `i` instead of `uint8`. use temporary variable for `operatorIds.length` and use it.

```suggestion
        uint256 numOperatorIds = operatorIds.length;
        for (uint256 i; i < numOperatorIds; ++i) {
```

if you don't much care about the readability, you can move `++i` to the end of the loop and put it to `unchecked` block.